go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/openshift-kni/oran-o2ims
 
-go 1.21
+go 1.22.0
+
+toolchain go1.22.8
 
```

Comment:
Do we still need the k8s.io/api and k8s.io/client-go replace directives? These are go-1.20 modules:
https://github.com/kubernetes/api/blob/v0.28.2/go.mod
https://github.com/kubernetes/client-go/blob/v0.28.2/go.mod

I guess we need to check the siteconfig operator dependencies here. Maybe we can update or remove the related replace directive completely?

The latest siteconfig operator is go-1.21, but still uses the 0.28.2 api and client-go.

Reply:
https://github.com/stolostron/siteconfig/pull/112

Reply:
I created a PR to remove the directive from siteconfig. However TALM has one of these directives, and it needs quite a bit of work to remove. It needs complicated code changes.
Maybe if move our APIs to a separate repo we could avoid these issues.

Reply:
You could do as we've done in this repo, creating go submodules for the exported APIs. The oran-hwmgr-plugin is able to import just the api/hardwaremanagement folder, for example, without then pulling in all the dependencies from the main oran-o2ims go.mod